src/libfetchers/git.cc

```diff
@@ -269,6 +269,10 @@ struct GitInputScheme : InputScheme
                     "lastModified",
                     haveCommits ? std::stoull(runProgram("git", true, { "-C", actualUrl, "log", "-1", "--format=%ct", "HEAD" })) : 0);
 
+                // Backward compatibility: set 'rev' to
+                // 0000000000000000000000000000000000000000 for a dirty tree.
+                input.attrs.insert_or_assign("rev", "0000000000000000000000000000000000000000");
```

Comment:
This should only be done for `fetchGit`.

Reply:
Do you want me to move this to the primop implementation? This is already the case for `git`-sources only.

Reply:
What I mean is that it shouldn't be done for `fetchTree`, even for Git inputs.

Reply:
Fixed. However there's now no `rev` field if only a url for fetchGit is specified.